pandas/io/pytables.py

```diff
@@ -1229,6 +1229,7 @@ def walk(self, where="/"):
     def get_node(self, key):
         """ return the node with the key or None if it does not exist """
         self._check_if_open()
+        _tables()
```

Comment:
huh? how does this do anything

Reply:
`_tables()` imports `tables` and sets `_tables_mod` equal to `tables`
without this line of code, `_tables_mod` is None, and then calling `_table_mod.exceptions.NoSuchNodeError` gives an AttributeError

Reply:
after adding this line of code, my package works fine, because the try except actually goes through
otherwise, I have to require pandas <=0.23.4 to avoid the AttributeError

Reply:
where exactly is this *every* set: ```_tables_mod ```

show an actual traceback

Reply:
its really unclear what is leading to this path, that's why an example of what you are calling is so important. we have a very extensive test suite and are not hitting this, so either you are doing something unexpected and there is a bug, or the api is being used incorrectly.

Reply:
#### Full stacktrace

<details>

Traceback (most recent call last):
  File "C:\Users\Tudor\Anaconda3\envs\supereeg_env\lib\site-packages\pandas\io\pytables.py", line 1160, in get_node
    return self._handle.get_node(self.root, key)
  File "C:\Users\Tudor\Anaconda3\envs\supereeg_env\lib\site-packages\tables\file.py", line 1644, in get_node
    node = where._v_file._get_node(nodepath)
  File "C:\Users\Tudor\Anaconda3\envs\supereeg_env\lib\site-packages\tables\file.py", line 1599, in _get_node
    node = self._node_manager.get_node(nodepath)
  File "C:\Users\Tudor\Anaconda3\envs\supereeg_env\lib\site-packages\tables\file.py", line 437, in get_node
    node = self.node_factory(key)
  File "C:\Users\Tudor\Anaconda3\envs\supereeg_env\lib\site-packages\tables\group.py", line 1181, in _g_load_child
    node_type = self._g_check_has_child(childname)
  File "C:\Users\Tudor\Anaconda3\envs\supereeg_env\lib\site-packages\tables\group.py", line 398, in _g_check_has_child
    % (self._v_pathname, name))
tables.exceptions.NoSuchNodeError: group ``/`` does not have a child named ``//locs``

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Users\Tudor\Anaconda3\envs\supereeg_env\lib\site-packages\supereeg\brain.py", line 638, in save
    dd.io.save(fname, bo, compression=compression)
  File "C:\Users\Tudor\Anaconda3\envs\supereeg_env\lib\site-packages\deepdish\io\hdf5io.py", line 584, in save
    filters=filters, idtable=idtable)
  File "C:\Users\Tudor\Anaconda3\envs\supereeg_env\lib\site-packages\deepdish\io\hdf5io.py", line 253, in _save_level
    store.put(group._v_pathname + '/' + name, level)
  File "C:\Users\Tudor\Anaconda3\envs\supereeg_env\lib\site-packages\pandas\io\pytables.py", line 889, in put
    self._write_to_group(key, value, append=append, **kwargs)
  File "C:\Users\Tudor\Anaconda3\envs\supereeg_env\lib\site-packages\pandas\io\pytables.py", line 1367, in _write_to_group
    group = self.get_node(key)
  File "C:\Users\Tudor\Anaconda3\envs\supereeg_env\lib\site-packages\pandas\io\pytables.py", line 1161, in get_node
    except _table_mod.exceptions.NoSuchNodeError:
AttributeError: 'NoneType' object has no attribute 'exceptions'


</details>

this stacktrace is also in #27253

Reply:
what is *before* this; how are you creating the HDFStrore, with what arguments, how are you calling it, etc.

Reply:
I did a little digging into deepdish, which is the package we're using for all our io, and here's what I could find

The HDFStore is created [here](https://github.com/uchicago-cs/deepdish/blob/01af93621fe082a3972fe53ba7375388c02b0085/deepdish/io/hdf5io.py#L35) in a subclass
and then [put](https://github.com/uchicago-cs/deepdish/blob/01af93621fe082a3972fe53ba7375388c02b0085/deepdish/io/hdf5io.py#L253) is called, which cascades to pandas `_write_to_group` then to `get_node` and then finally throws the AttributeError

Reply:
so we have *0* tests on subclassing HDFStore; I actually don't think this is a good idea. However you should change your package to call _tables() in your init then. Unless you have a reproducible example on that is self contained, this is not a bug in pandas.